src/util/mod.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! utility functions
+
+use pnet;
+use std::io::{Read, BufRead, BufReader};
+use std::fs::File;
+use std::path::Path;
+use std::time::Duration;
+use errors::*;
+use retry;
+
+fn key_lookup_line(delim: char, key: &str, line: &str) -> Option<String> {
+    match line.find(delim) {
+        Some(index) => {
+            let (k, val) = line.split_at(index+1);
+            if k != format!("{}{}", key, delim) {
+                None
+            } else {
+                Some(val.to_owned())
+            }
+        }
+        None => None,
+    }
+}
+
+pub fn key_lookup_reader<R: Read>(delim: char, key: &str, reader: R) -> Result<Option<String>> {
+    let contents = BufReader::new(reader);
+
+    for l in contents.lines() {
+        let l = l?;
+        if let Some(v) = key_lookup_line(delim, key, &l) {
+            return Ok(Some(v));
+        }
+    }
+    Ok(None)
+}
+
+pub fn key_lookup(delim: char, key: &str, contents: &str) -> Option<String> {
```

Comment:
Building this results in warnings:

```
warning: function is never used: `key_lookup`
  --> src/util/mod.rs:51:1
   |
51 | / pub fn key_lookup(delim: char, key: &str, contents: &str) -> Option<String> {
52 | |     for l in contents.lines() {
53 | |         if let Some(v) = key_lookup_line(delim, key, l) {
54 | |             return Some(v)
...  |
57 | |     None
58 | | }
   | |_^
   |
   = note: #[warn(dead_code)] on by default
```

Reply:
There are a few other clippy warnings as well. I think we can tackle that after merging and setting up travis-ci for it as well. I already did for the other crates we have, and I'm planning to do it here as well.